org.lflang/src/org/lflang/generator/c/CDockerGenerator.java

```diff
@@ -1,38 +1,106 @@
 package org.lflang.generator.c;
 
 import org.eclipse.xtext.xbase.lib.IterableExtensions;
+import org.lflang.FileConfig;
 import org.lflang.TargetConfig;
 import org.lflang.generator.DockerGeneratorBase;
 import org.lflang.util.StringUtil;
 
 /**
  * Generate the docker file related code for the C and CCpp target.
  *
- * @author{Edward A. Lee <eal@berkeley.edu>}
  * @author{Hou Seng Wong <housengw@berkeley.edu>}
  */
 public class CDockerGenerator extends DockerGeneratorBase {
     private boolean CCppMode;
     private TargetConfig targetConfig;
     private final String defaultBaseImage = "alpine:latest";
 
+    /**
+     * The interface for data from the C code generator.
+     */
+    public class CGeneratorData implements GeneratorData {
+        /**
+         * The name of the LF module in CGenerator.
+         * Typically, this is "fileConfig.name + _ + federate.name"
+         * in federated execution and "fileConfig.name" in non-federated
+         * execution.
+         */
+        private String lfModuleName;
+        /**
+         * The value of "currentFederate.name" in CGenerator.
+         */
+        private String federateName;
```

Comment:
Same concern here about federation lingo.

Reply:
this is OK since `CGeneratorData` is specific to docker file generation in the C target, and based on the current design of `CGenerator` the value `federate.name` is used to generate the docker file.